packages/collector/src/tracing/instrumentation/process/childProcess.js

```diff
@@ -57,6 +60,76 @@ function shimSpawn(original) {
   };
 }
 
+const bullMasterProcessMatch = /bull\/lib\/process\/master\.js/;
+
+function shimFork(original) {
+  return function() {
+    // args: modulePath, args, options
+    const _args = Array.prototype.slice.call(arguments);
+    const modulePath = _args[0];
+    const args = _args[1];
+
+    if (typeof modulePath === 'string' && bullMasterProcessMatch.test(modulePath) && args && args.execArgv) {
+      if (!selfPath.immediate) {
+        logger.warn(
+          "Detected a child_process.spawn of 'Bull processor', but the path to @instana/collector is not available, " +
+            'so this Bull processor instance will not be instrumented.'
+        );
+      } else {
+        logger.debug(
+          `Detected a child_process.spawn of Bull, instrumenting it by adding --require ${selfPath.immediate}.`
+        );
+
+        process.env.INSTANA_AGENT_UUID = processIdentityProvider.getFrom().h;
+        args.execArgv.unshift(selfPath.immediate);
+        args.execArgv.unshift('--require');
+      }
+
+      /** @type {import('node:child_process').ChildProcess} */
+      const childProcess = original.apply(this, _args);
+
+      // Retrieve the entry span created by bull.js#instrumentedProcessJob.
+      const originalChildProcessSend = childProcess.send;
+      childProcess.send = function(message) {
+        const entrySpan = cls.getCurrentSpan();
+        if (
+          //
+          message && //
+          message.cmd === 'start' &&
+          message.job &&
+          typeof message.job === 'object' &&
+          message.job.opts &&
+          entrySpan &&
+          entrySpan.k === 1 &&
+          processIdentityProvider
```

Comment:
This check makes no sense, because we access `processIdentityProvider` already way before we get here (line 83). I probably added it and we refactored this part later.